cfgov/v1/models/snippets.py

```diff
@@ -39,6 +49,93 @@ def get_by_title_slug(self, title, slug):
             hash=hashlib.md5(title + ';;' + slug).hexdigest())
 
 
+class ResourceTag(TaggedItemBase):
+    content_object = ParentalKey('v1.Resource', related_name='tagged_items')
+
+
+class TaggableSnippetManager(models.Manager):
+    def filter_by_tags(self, tags):
+        snippets = self.all()
+        for tag in tags or []:
+            snippets = snippets.filter(tags__name=tag)
+
+        return snippets
+
+
+@register_snippet
+class Resource(ClusterableModel):
+    title = models.CharField(max_length=255)
+    desc = RichTextField(verbose_name='Description', blank=True)
+
+    thumbnail = models.ForeignKey(
+        'v1.CFGOVImage',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
```

Comment:
Sounds like it would be better. I just copied this code from elsewhere; should we always be using PROTECT instead of SET_NULL?